### PR TITLE
fix: changed CD so it only runs from main

### PR DESCRIPTION
**Pull request by ktavean**

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+            - main
+    pull_request:
+        branches:
+            - main
+    workflow_dispatch:
 
 jobs:
     cd:
         runs-on: ${{ matrix.os }}
+        if: github.ref == 'refs/heads/master'
```
Comment (vladutcornel) [inline .github/workflows/cd.yml:15]: We probably don't need this, if we have the rules above.

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 
-on: [push, pull_request, workflow_dispatch]
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+    workflow_dispatch:
```
Comment (vladutcornel) [inline .github/workflows/cd.yml:10]: I think this is not valid YAML. If we can't put anything here, we can remove `workflow_dispatch`.

**Review by vladutcornel — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (ktavean) [reply .github/workflows/cd.yml]: My bad, I forgot to remove this.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (ktavean) [reply .github/workflows/cd.yml:10]: According to the reference, it seems to be valid: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-activity-types-and-filters-with-multiple-events

I believe workflow_dispatch falls under the category of an event without configuration

Comment (vladutcornel) [reply .github/workflows/cd.yml:10]: Ok. Let's see if this works.

**Pull request merged**